xbmc/utils/StringUtils.cpp

```diff
@@ -1066,14 +1067,14 @@ int64_t StringUtils::AlphaNumericCompare(std::wstring_view left, std::wstring_vi
     {
       ld = l;
       lnum = *ld++ - L'0';
```

Comment:
Suggestion: declare lnum here
```suggestion
      int64_t lnum{*ld++ - L'0'};
```
so it's not left uninitialized on declaration, and available in the smallest required scope. Same for rnum.
lsym and rsym could have a similar treatment (and be const)